tests/automatic_differentiation/ad_argon.i

```diff
@@ -754,8 +754,22 @@ dom0Scale=1e-3
     electron_density = 'em'
     include_electrons = true
     file_location = 'argon_chemistry_rates'
-    equation_constants = 'Tgas e_temp'
-    equation_values = '300 34800'
+    equation_constants = 'Tgas'
+    equation_values = '300'
+
+    # For function-based reactions (e.g. those whose rate coefficients are
+    # enclosed in curly braces, {}), the e_temp auxiliary variable is
+    # named as a variable. Note that while e_temp is an AuxVariable, it does
+    # depend on nonlinear variables and therefore should contribute to the
+    # jacobian of any functions which depend on it.
+    # At the moment this jacobian contribution is not included, but the
+    # contribution is minimal compared to advection and diffusion so we
+    # can get away with a NEWTON solver in this case.
+    #
+    # For more complex reaction networks with many functions that depend on
+    # electron temperature, it may be necessary to use PJFNK instead of
+    # NEWTON.
```

Comment:
Just a comment - I really appreciate the extended description comments / caveats here. Will be a huge help for new users!